test/parallel/test-process-env-flags.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const assert = require('assert');
```

Comment:
Not a fan of this test as it just validates the same list and we'd just have to keep it synced it here twice whenever a flag is added. We might want to just check certain flags that we consider "crucial" (like `--inspect`) exist?

Reply:
Or alternatively, grep the options from `node --help` and assert that all of the options from the binary are in `process.allowedEnvironmentNodeFlags` or something like that? (Taking the disallowed ones into account)

Reply:
I'm feeling like matching them exactly is the most straightforward thing to do, as there may be some debate (which I don't want to enter at this point) over what's "crucial" or not, and whether new flags should get added.

Grepping `node --help` seems a bit painful, but moreover reflects a superset of options, including those unsupported by `NODE_OPTIONS`.  We can hardcode exceptions for now (which will be yet another hardcoded list, including stuff like `--help` and `-e`), but any future flags *also* unsupported by `NODE_OPTIONS` would need to be added to the exceptions list.

In summary, I'd like to just leave these as-is for now, until there's an obvious big win in doing it another way.